### PR TITLE
chore: upgrade nx 17

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

Upgrade Nx to latest version

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> One day they will all understand

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] my-env - chore: upgrade nx - 2023-10-25T16:09:36Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/140bbfb9-a14b-48b2-ac11-ab586c9265f6/logs)

File: package.json [hunk 1]
```diff
@@ -122,9 +122,9 @@
     "chance": "^1.1.8",
     "css-loader": "^6.8.1",
     "cypress": "^9.1.0",
-    "eslint": "^8.49.0",
-    "eslint-config-prettier": "8.5.0",
-    "eslint-plugin-cypress": "^2.10.3",
+    "eslint": "8.46.0",
```
Comment (RemiBonnet) [inline package.json:125]: `eslint` version downgrade?

Comment (ctjhoa) [reply package.json]: I've done the upgrade manually to avoir regression
https://github.com/Qovery/console/pull/921/commits/1b60c11403b08e427ffa6a6050fb91b57eacbb22

File: libs/shared/ui/tsconfig.storybook.json [hunk 2]
```diff
@@ -5,27 +5,10 @@
     "outDir": ""
   },
   "files": [
-    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
+    "../../../node_modules/@nx/react/typings/styled-jsx.d.ts",
```
Comment (RemiBonnet) [inline libs/shared/ui/tsconfig.storybook.json:8]: Maybe a mistake it's weird to have a different path compared with the bottom paths, no? 

If it's generate by Nx ok for me 👍

Comment (ctjhoa) [reply libs/shared/ui/tsconfig.storybook.json:8]: It doesn't seems to have impact on storybook but it's weird you're right. I'll check on fresh project

Comment (RemiBonnet) [reply libs/shared/ui/tsconfig.storybook.json:8]: Yes I tried the Storybook locally and it works

Comment (ctjhoa) [reply libs/shared/ui/tsconfig.storybook.json:8]: Have align the other `d.ts` this is what is generated by Nx from scratch
https://github.com/Qovery/console/pull/921/commits/a33b0aa5f5c14bfb50c539deead2543eab673066

**Pull request merged**